xain_fl/fl/coordinator/controller.py

```diff
@@ -7,56 +7,60 @@
-            all participant ids will be selected.
+            all participant ids will be selected. It must be in the (0.0, 1.0] interval.
     """
 
-    def __init__(
-        self, participant_ids: List[str], fraction_of_participants: float = 1.0
-    ) -> None:
-        self.participant_ids: List[str] = participant_ids
+    def __init__(self, fraction_of_participants: float = 1.0) -> None:
         self.fraction_of_participants: float = fraction_of_participants
-        self.num_ids_to_select: int = self.get_num_ids_to_select()
 
-    def get_num_ids_to_select(self) -> int:
+    def get_num_ids_to_select(self, participant_ids: List[str]) -> int:
         """Calculates how many participant ids need to be selected.
 
+        Args:
+            participant_ids (:obj:`list` of :obj:`str`): The list of IDs of the
+                all the available participants, to calculate its length.
+
         Returns:
             :obj:`int`: Number of participant ids to be selected
         """
-        raw_num_ids_to_select = (
-            len(self.participant_ids) * self.fraction_of_participants
-        )
+        raw_num_ids_to_select = len(participant_ids) * self.fraction_of_participants
         max_valid_value = max(1, np.ceil(raw_num_ids_to_select))
-        minimum_valid_value = min(len(self.participant_ids), max_valid_value)
+        minimum_valid_value = min(len(participant_ids), max_valid_value)
         return int(minimum_valid_value)
 
     @abstractmethod
-    def select_ids(self) -> List[str]:
-        """Returns the selected indices of next round
+    def select_ids(self, participant_ids: List[str]) -> List[str]:
+        """Returns the selected indices of next round.
+
+        Args:
+            participant_ids (:obj:`list` of :obj:`str`): The list of IDs of all the
+                available participants, a subset of which will be selected.
 
         Returns:
-            :obj:`list` of :obj:`str`: Unordered list of selected ids
+            :obj:`list` of :obj:`str`: List of selected participant IDs
         """
         raise NotImplementedError("not implemented")
 
 
 class RandomController(Controller):
-    def select_ids(self) -> List[str]:
+    def select_ids(self, participant_ids: List[str]) -> List[str]:
         """Randomly samples self.num_ids_to_select from the population of participants_ids,
         without replacement.
 
+        Args:
+            participant_ids (:obj:`list` of :obj:`str`): The list of IDs of all the
+                available participants, a subset of which will be selected.
+
         Returns:
-            :obj:`list` of :obj:`str`: List of selected participant ID's
+            :obj:`list` of :obj:`str`: List of selected participant IDs
         """
-        return np.random.choice(
-            self.participant_ids, size=self.num_ids_to_select, replace=False
-        )
+        num_ids_to_select = self.get_num_ids_to_select(participant_ids)
```

Comment:
@PanicButtonPressed I think it is enough if we just change the line here 
 ```suggestion
         num_ids_to_select = self.get_num_ids_to_select(len(participant_ids))
```